django/db/models/options.py

```diff
@@ -972,11 +971,15 @@ def total_unique_constraints(self):
     @cached_property
     def _property_names(self):
         """Return a set of the names of the properties defined on the model."""
-        names = []
-        for name in dir(self.model):
-            attr = inspect.getattr_static(self.model, name)
-            if isinstance(attr, property):
-                names.append(name)
+        names = set()
+        for klass in self.model.__mro__:
+            names.update(
+                {
+                    name
+                    for name, value in klass.__dict__.items()
+                    if isinstance(value, property)
+                }
+            )
```

Comment:
You can avoid all the call overhead for `.update()`:

```suggestion
            names |= {
                name
                for name, value in klass.__dict__.items()
                if isinstance(value, property)
            }
```

The code is cleaner and also slightly faster:

```pycon
In [1]: %%timeit x = set()
   ...: for i in range(100):
   ...:     x.update({"a", "b", "c"})
   ...: 
18.5 µs ± 2.77 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

In [2]: %%timeit x = set()
   ...: for i in range(100):
   ...:     x |= {"a", "b", "c"}
   ...: 
13.1 µs ± 582 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

Reply:
I always forget the set operators, thanks! 👍

Reply:
It seems you forgot to commit Nick's suggestion.

Reply:
I forgot to push it 🤦